src/v/kafka/server/group_router.cc

```diff
@@ -81,4 +229,93 @@ group_router::delete_groups(std::vector<group_id> groups) {
       });
 }
 
+ss::future<described_group> group_router::describe_group(kafka::group_id g) {
+    auto m = shard_for(g);
+    if (!m) {
+        return ss::make_ready_future<described_group>(
+          describe_groups_response::make_empty_described_group(
+            std::move(g), error_code::not_coordinator));
+    }
+    return with_scheduling_group(
+      _sg, [this, g = std::move(g), m = std::move(m)]() mutable {
+          return get_group_manager().invoke_on(
+            m->second,
+            _ssg,
+            [g = std::move(g),
+             ntp = std::move(m->first)](group_manager& mgr) mutable {
+                return mgr.describe_group(ntp, g);
+            });
+      });
+}
+group::join_group_stages
+group_router::join_group(join_group_request&& request) {
+    return route_stages(std::move(request), &group_manager::join_group);
```

Comment:
Note that I have only modified `route_stages` not `route`. 

This seems to line up roughly with what we want: we want this special handling for requests with a "long poll" second part, like join group and also sync group. I don't think we need it for the other request types which are generally "straight though", i.e., do not involve a long poll that interacts with the consumer-side state machine.

I do wonder if it is appropriate to apply this logic to `offset_commit` which also uses two stages?